storage/google/cloud/storage/bucket.py

```diff
@@ -345,6 +345,14 @@ def user_project(self):
         """
         return self._user_project
 
+    @property
+    def query_params(self):
+        """Default query parameters."""
+        params = {}
+        if self.user_project is not None:
+            params["userProject"] = self.user_project
+        return params
+
     def blob(self, blob_name, chunk_size=None, encryption_key=None, kms_key_name=None):
         """Factory constructor for blob object.
```

Comment:
I think defining the `generation` would be helpful here as well. Given I pass around the generation number and then use it to create a blob instance without performing a `GET`. 

This could be used with for example other operations: `patch()`, `update()`, `rewrite()` for example.